### PR TITLE
Fix version orders

### PR DESCRIPTION
**Pull request by JKarlavige**
### Issue
With the current versioning, `v1.10` is considered less than `v1.9`. This adjusts the versioning to correctly sort the version numbers by major, minor and patch versions.

## Preview

Verify the `VersionBlock` components correctly show/hide based on the `firstVersion` & `lastVersion` props set:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/docs/version-testing

Verify this page hides from the sidebar for versions after `v1.8`:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/resource-configs/target_database

Verify `Project configs` sidebar category is only visible for `v1.8` and up:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/dbt_project.yml

To quickly test the new `sortVersions` method, drop the following code into the versionBlock file and visit the `/version-testing` page above. There is a console.log within `sortVersions` which will show the sorted array of versions.
```
sortVersions([
  "1.11",
  "1.7",
  "2.1",
  "1.8",
  "1.6",
  "2.2.1",
  "1.10",
  "2.0.1",
  "1.9",
  "1.9.1",
  "1.10.1",
  "2.0",
]);
```

## TODO Before Merge
- [ ] Delete demo `version-testing` page
- [ ] Remove console.log from `sortedVersions`
- [ ] Revert versions in `dbt-versions`
- [ ] Remove demo `Project configs` versioned category in `dbt-versions`

### CODE REVIEW

File: contributing/single-sourcing-content.md
Comment (JKarlavige) [inline contributing/single-sourcing-content.md:1]: Functionality not used, confirmed with Docs team we can remove the `Var` component: https://dbt-labs.slack.com/archives/C02NCQ9483C/p1726157412409069

File: website/dbt-versions.js [hunk 1]
```diff
@@ -43,11 +55,11 @@ exports.versions = [
  */
 exports.versionedPages = [
   {
-    "page": "/reference/resource-configs/target_database",
+    "page": "reference/resource-configs/target_database",
     "lastVersion": "1.8",
   },
   {
-    "page": "/reference/resource-configs/target_schema",
+    "page": "reference/resource-configs/target_schema",
```
Comment (JKarlavige) [inline website/dbt-versions.js]: Keep these changes, page paths should not have leading forward slash. Remove all other changes in this file.

File: website/docs/docs/version-testing.md
Comment (JKarlavige) [inline website/docs/docs/version-testing.md:1]: Delete this page before merge

File: website/src/utils/sort-versions.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+   
+    // Sort by patch version
+    if (aPatch - bPatch < 0) { return -1; }
+    if (aPatch - bPatch > 0) { return 1; } 
+    
+    // If reached, a & b are equal
+    return 0
+  });
+
+  console.log('sortedVersions', sortedVersions)
```
Comment (JKarlavige) [inline website/src/utils/sort-versions.js]: Remove this console before merge

**Comment by john-rock** [comment]
Hey @JKarlavige nice work on this, qq on this page https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/resource-configs/target_database

Confirmed it does get hidden from the sidebar after `v1.8` but is this the correct copy that is suppose to be showing if the page is accessed when the version is set > `v1.8`?
image

Specifically the "You should upgrade to 0 or later if you want to use this feature."

**Comment by JKarlavige** [comment]
> Hey @JKarlavige nice work on this, qq on this page https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/resource-configs/target_database
> 
> Confirmed it does get hidden from the sidebar after `v1.8` but is this the correct copy that is suppose to be showing if the page is accessed when the version is set > `v1.8`? image
> 
> Specifically the "You should upgrade to 0 or later if you want to use this feature."

Nice catch, it should definitely not say that. Will work on fixing that!

**Comment by JKarlavige** [comment]
@john-rock Updated the version warning banner to exclude the "first available in version.." part of the message if a `firstVersion` prop isn't set for that versioned page. Previews below:

Verify version banner no longer shows incorrect upgrade recommendation line:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/resource-configs/target_database

Select `v1.7` or lower and verify version banner shows with correct message:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/global-configs/indirect-selection

Verify all pages not listed in [versionedPages array](https://github.com/dbt-labs/docs.getdbt.com/blob/a3edaf54a1e02c989ff9917c96fb578d8e222731/website/dbt-versions.js#L44) do not show version warning banner:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/docs/introduction

File: website/src/utils/sort-versions.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    const aSegments = a?.split(".", 3);
+    const bSegments = b?.split(".", 3);
+
+    // Store each version part in variable to help readability below
+    const aMajor = aSegments[0] || "0"
+    const bMajor = bSegments[0] || "0"
+    const aMinor = aSegments[1] || "0"
+    const bMinor = bSegments[1] || "0"
+    const aPatch = aSegments[2] || "0"
+    const bPatch = bSegments[2] || "0"
```
Comment (john-rock) [inline website/src/utils/sort-versions.js:25]: The lengths we go to with javascript 🤣 nice work on this util.

**Review by john-rock — APPROVED** [review]
This lgtm. I'm no longer seeing the incorrect warning message if `firstVersion` is not set.

**Pull request merged**